pkg/shared/utils/arrays/generic_array_utils.go

```diff
@@ -1,12 +1,86 @@
 package arrays
 
+const NotFound = -1
```

Comment:
Could we use a more specific name?

Right now the usage from client side would be `arrays.NotFound`. What is that is not found? Maybe `ElementNotFound`, `SliceElementNotFound` or something similar?
Does this constant need to be exported?

Reply:
The idea behind exporting it was to allow the user to do something like
```go

if idx, val := arrays.Find(.....); idx == array.NotFound {
...
}
```